[Misc] add dummy decorator when nvtx is not available

hijeffwu opened the pull request:
The PR is for：

- add dummy decorator when nvtx is not available

test code:
```
import sys
sys.path.insert(0, 'LMCache')
from lmcache.utils import _lmcache_nvtx_annotate

@_lmcache_nvtx_annotate
def test_func():
    return 'test successful'

result = test_func()

print(f'test result: {result}')
```

#python3.10 test.py
test result: test successful



@KuntaiDu, @ApostaC @YaoJiayi

motorma-enflame committed:
Stub implementation for the platforms that nvtx is not supported

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

motorma-enflame committed:
return original function when nvtx is not available.

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

Commit:
fix a type.

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

Commit:
add dummy decorator when nvtx is not available

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

Commit:
remove a blank line

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

hijeffwu commented:
@gemini-code-assist

hijeffwu commented:
/gemini review

hijeffwu commented on lmcache/utils.py:
```diff
@@ -11,7 +11,15 @@
+        """Dummy decorator when nvtx is not available."""
+        def decorator(func):
+            return func
+        return decorator
```
Overly complex and not feasible for this context.

hickeyma reviewed (changes requested):
Thanks @hijeffwu for pushing this PR.

Why would we want to use a dummy decorator? In what scenario would `nvtx` be unavailable?

hijeffwu commented:
> Thanks @hijeffwu for pushing this PR.
> 
> Why would we want to use a dummy decorator? In what scenario would `nvtx` be unavailable?

We are using non-NVIDIA AI accelerator cards based on a specialized architecture, which do not provide NVTX.

hickeyma reviewed (approved):
LGTM, thanks @hijeffwu

hickeyma commented on lmcache/utils.py:
```diff
@@ -306,6 +312,10 @@ def _get_color_for_nvtx(name):
+    if not _NVTX_AVAILABLE:
+        # return original function when nvtx is not available
+        return func
+
```
Ignore

hickeyma reviewed (commented):
@hijeffwu Can you fix the code quality checks? Advise enabling pre-commit hooks: https://docs.lmcache.ai/developer_guide/contributing.html#coding-style

ApostaC reviewed (approved):
Thanks for the contribution! Looks good to me. 

Please fix the code quality issue and we are happy to merge it.

Commit:
fix the code quality checks.

```
LMCache# pre-commit run --all-files
Check SPDX headers.......................................................Passed
isort....................................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
codespell................................................................Passed
clang-format.............................................................Passed
mypy.....................................................................Passed
```

Signed-off-by: Jeff Wu <jeff.wu@enflame-tech.com>

hijeffwu commented:
> @hijeffwu Can you fix the code quality checks? Advise enabling pre-commit hooks: https://docs.lmcache.ai/developer_guide/contributing.html#coding-style

Hi @hickeyma , had fixed. 
https://github.com/LMCache/LMCache/pull/1502/commits/67462fab077f60622e31af625183f95c2e53164a

YaoJiayi reviewed (approved):
lgtm

hickeyma reviewed (commented):
Rebased the PR with latest codebase as unit tests were failing and it did not seem to be caused by the PR.

CI is passing now, so merging.
